sched/signal/sig_action.c

```diff
@@ -53,10 +53,13 @@
 static FAR sigactq_t *nxsig_alloc_action(void)
 {
   FAR sigactq_t *sigact;
+  irqstate_t flags;
 
   /* Try to get the signal action structure from the free list */
 
+  flags = spin_lock_irqsave(NULL);
```

Comment:
Do we need the global IRQ lock here?  I think a private lock would work fine.

Reply:
Hmmm... I'm not sure, `g_sigfreeacation` is global and accessed elsewhere.  Can we add the same lock for line 67?

Reply:
BTW can't we move `nxsig_alloc_actionblock` to `sig_action.c`?

Reply:
done